dotcom-rendering/src/web/components/HydrateOnce.tsx

```diff
@@ -51,6 +51,7 @@ export const HydrateOnce = ({ rootId, children, waitFor = [] }: Props) => {
 	if (!element) return null;
 	start();
 	ReactDOM.hydrate(children, element, () => {
+		element.setAttribute('data-cy-hydrated', 'true');
```

Comment:
Is this too Cypress-specific? If we add this to all elements that are hydrated, maybe we want to make it more Guardian-specific, like `data-gu-hydrated`?

Reply:
Ha. Yeah, I went back and forth on this. In the end I realised I can't think of a use case for where this would be used outside of Cypress so it felt right to use the `data-cy-*` convention. But I'm still on the fence. One side of my brain is telling me, but you might want to use this elsewhere in future! And the other half is telling me to only make code changes based on what you need right now 🤷 

All that said, I think that you make a valid point which I haven't considered which is that we're _always_ adding this attribute, not just for you tube elements, so lets go with `data-gu-hydrated` 👍

Reply:
I feel the struggle, but I for one would find it helpful to be able to tell from bringing up the inspector whether an element has been hydrated. Seeing `gu` makes me feel at home and that it is something _we intended_. Seeing `cy` would not give me the same _feeling_, or make me think it’s done by Cypress.

Reply:
I think you're right. As an aside, the reason we need to use a marker like this is because we're implementing 'islands' of interactivity, instead of hydrating the whole page. If we just did the classic `React.hydrate(TheWholePage)` then you'd be able to tell when hydration has completed with a simple `useEffect` / checking for the UI changes. But with islands things complete at different points and this was causing flakiness in the tests.